authn/authenticate.go

```diff
@@ -25,6 +25,8 @@ import (
 	"golang.org/x/oauth2"
 )
 
+type AuthContextKey string
```

Comment:
(not sure where my comment went) Let's get in the habit of adding go-docs to any exported type/method/function that we add or modify in the code.

Reply:
Updated